### PR TITLE
Flow setup guide

### PR DESCRIPTION
**Pull request by mayank4kathuria**
This is a developer's guide for easy setting up of FLOW tool from scratch.

### CODE REVIEW

File: docs/flow-setup-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Flow setup Tutotrial
+
+Flow is a static-type checker for Javascript code
+
+## The steps to install Flow correctly from scratch are
+
+####  Installation tip : install flow in the root folder of zulip-mobile
```
Comment (borisyankov) [inline docs/flow-setup-tutorial.md]: That is not desirable. We want it either installed globally, or via `package.json` in `./node_packages`.

File: docs/flow-setup-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+####  package installer used - yarn
+
+## Installing flow steps
+
+1. install babel compiler with command line interface,
```
Comment (borisyankov) [inline docs/flow-setup-tutorial.md]: Why is that needed?

File: docs/flow-setup-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	run command <$yarn add --dev babel-cli babel-preset-flow>
+2. change the present from "react-native" to "flow"
+>	{
+>	"present": ["flow"]
```
Comment (borisyankov) [inline docs/flow-setup-tutorial.md]: Why is that? We either should change it ourselves or not suggest it at all?

File: docs/flow-setup-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+>	}
+3. Install flow version 0.65 for zulip using <$yarn add --dev flow-bin@0.65>
```
Comment (borisyankov) [inline docs/flow-setup-tutorial.md]: The Flow version suggested should be gotten from `.flowconfig` - it is not hardcoded to `0.65`.

File: docs/flow-setup-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+4. type <$flow init> 
+5. Type flow server by typing <$yarn run flow> to check if flow is correctly installed
+
+## Uninstalling flow
```
Comment (borisyankov) [inline docs/flow-setup-tutorial.md]: I don't think we need to give information how to uninstall.

Comment (mayank4kathuria) [reply docs/flow-setup-tutorial.md]: in case the developer isn't running the correct dev environment then flow doesn't work as expected. In that case a complete uninstall and reinstall of flow is required, that's why it is there.

File: docs/flow-setup-tutorial.md (hunk 4 above)
Comment (mayank4kathuria) [reply docs/flow-setup-tutorial.md]: noted

File: docs/flow-setup-tutorial.md (hunk 3 above)
Comment (mayank4kathuria) [reply docs/flow-setup-tutorial.md]: The flow doc specifies to set the "present" variable to flow in the .babelrc , i checked the file and it had react-native present, i added flow as another value but it didn't worked so i experimented and replaced react-native with flow and it worked fine.

Comment (mayank4kathuria) [reply docs/flow-setup-tutorial.md]: The .babelrc file is generated after installing babel-cli so we need to give information to the developer about the change.

File: docs/flow-setup-tutorial.md (hunk 2 above)
Comment (mayank4kathuria) [reply docs/flow-setup-tutorial.md]: The flow doc says this " Babel is a compiler for JavaScript code that has support for Flow. Babel will take your Flow code and strip out any type annotations. " , it's the first step of installing flow.

File: docs/flow-setup-tutorial.md (hunk 1 above)
Comment (mayank4kathuria) [reply docs/flow-setup-tutorial.md]: I found this line "Flow works best when installed per-project with explicit versioning rather than globally." in the flow's documentation , that's why i added that line.

File: docs/flow-setup-tutorial.md (hunk 3 above)
Comment (mayank4kathuria) [reply docs/flow-setup-tutorial.md]: update: it is working with react-native as it's value as well. Let me perform some other checks as well before final reporting.

File: docs/flow-setup-tutorial.md (hunk 1 above)
Comment (borisyankov) [reply docs/flow-setup-tutorial.md]: Ok, that is reasonable. Then we can make it run locally. We don't need to add instructions but just implement it.

Comment (borisyankov) [reply docs/flow-setup-tutorial.md]: We are currently installing the correct Flow version. The only thing a user needs to do is `npm run test:flow`

File: docs/flow-setup-tutorial.md (hunk 2 above)
Comment (borisyankov) [reply docs/flow-setup-tutorial.md]: Babel is already installed with React Native

File: src/utils/timing.js [hunk 6]
```diff
@@ -1,56 +1,57 @@
...
-
-export default {
-  add,
-  startGroup,
-  start,
-  end,
-  endGroup,
-  log,
-};
+/* @flow */
```
Comment (borisyankov) [inline src/utils/timing.js]: This file is included incorrectly here?

File: docs/flow-setup-tutorial.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+## Installing flow steps
+
+1. install babel compiler with command line interface,
+	run command <$yarn add --dev babel-cli babel-preset-flow>
+2. change the present from "react-native" to "flow"
+>	{
+>	"present": ["flow"]
+>	}
+3. Install flow version 0.65 for zulip using <$yarn add --dev flow-bin@0.xx> where xx is the flow version defined in .flowconfig file.
+4. type <$flow init>
```
Comment (borisyankov) [inline docs/flow-setup-tutorial.md]: This is not needed. This is setting up Flow in a new project. We have already done that.

File: docs/flow-setup-tutorial.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+## Installing flow steps
+
+1. install babel compiler with command line interface,
+	run command <$yarn add --dev babel-cli babel-preset-flow>
+2. change the present from "react-native" to "flow"
+>	{
+>	"present": ["flow"]
+>	}
+3. Install flow version 0.65 for zulip using <$yarn add --dev flow-bin@0.xx> where xx is the flow version defined in .flowconfig file.
```
Comment (borisyankov) [inline docs/flow-setup-tutorial.md]: `package.js` already has this installed.

File: docs/flow-setup-tutorial.md (hunk 1 above)
Comment (mayank4kathuria) [reply docs/flow-setup-tutorial.md]: i have written this guide for yarn users , should i update for npm users as well ?

**Comment by gnprice** [comment]
Hi @mayank4kathuria !

First, an important GitHub tip: please keep the changes in separate PRs separate. It looks like the issue is that you've used your `master` branch for both this PR and #2125 . Instead, you should make a separate branch for each set of changes (a "feature branch"), one branch per PR. Take a look at [our Git Guide](https://zulip.readthedocs.io/en/latest/git/using.html).

These instructions are not what you should be doing to set up Flow for Zulip.
* Our complete intended setup is to follow [our Developer Guide](https://github.com/zulip/zulip-mobile/blob/master/docs/developer-guide.md).
* Once you've done that, you should be able to run Flow with just `yarn run flow`. (Or `yarn run test:flow`, or `npm run test:flow`.)

Did you follow the setup in the Developer Guide, and did something not work for you? If you had any trouble, I'd love to debug it -- please chat in the #mobile stream on chat.zulip.org.
